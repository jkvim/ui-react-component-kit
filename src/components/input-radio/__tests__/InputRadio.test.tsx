import * as React from 'react';
import { mount } from 'enzyme';
import { InputRadio } from '../';
import toJSON from 'enzyme-to-json';

describe('InputRadio', () => {
  let wrapper;

  const callback = jest.fn();
  const label = 'Select this radio to confirm that you agree with the following statements';

  beforeEach(() => {
    wrapper = mount(
      <InputRadio
        id="testId"
        label={label}
        onChange={callback}
      />
    );
  });

  it('should match snapshot', () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('renders input tag', () => {
    expect(wrapper.find('input')).toHaveLength(1);
  });

  it('should render checkbox unticked', () => {
    const checkbox = wrapper.find('input');
    expect(checkbox.props().checked).toBeFalsy();
  });

  it('renders label', () => {
    expect(wrapper.find('label')).toHaveLength(1);
  });

  it('renders label with passed in label text', () => {
    expect(wrapper.find('label').text()).toBe(label);
  });

  describe('onChange callback', () => {
    test('should call onChange callback when provided', () => {
      expect(callback).not.toBeCalled();
      wrapper.find('input').simulate('change');
      expect(callback).toBeCalled();
    });
  });
});
