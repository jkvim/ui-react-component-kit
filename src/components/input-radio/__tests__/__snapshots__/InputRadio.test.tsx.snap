// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputRadio should match snapshot 1`] = `
<InputRadio
  id="testId"
  label="Select this radio to confirm that you agree with the following statements"
  onChange={[MockFunction]}
>
  <styled.div>
    <div
      className="sc-gzVnrw kDUAmj"
    >
      <styled.input
        checked=""
        hasError={false}
        id="testId"
        innerRef={[Function]}
        onChange={[Function]}
        theme={
          Object {
            "color": Object {
              "borderColor": "#333",
              "errorColor": "#9f173f",
            },
          }
        }
        type="radio"
      >
        <input
          checked=""
          className="sc-EHOje fyRfII"
          id="testId"
          onChange={[Function]}
          type="radio"
        />
      </styled.input>
      <styled.label
        hasError={false}
        htmlFor="testId"
        theme={
          Object {
            "color": Object {
              "color": "#333",
              "errorColor": "#9f173f",
            },
          }
        }
      >
        <label
          className="sc-htpNat nmRTs"
          htmlFor="testId"
        >
          Select this radio to confirm that you agree with the following statements
        </label>
      </styled.label>
      <styled.div
        theme={
          Object {
            "color": Object {
              "errorColor": "red",
            },
          }
        }
      >
        <div
          className="sc-dnqmqq gpxgKf"
        />
      </styled.div>
    </div>
  </styled.div>
</InputRadio>
`;
